ci/infra/testrunner/utils/command.py

```diff
@@ -0,0 +1,44 @@
+class CommandResult:
+    """
+    Embeds the result of running a command for easier testing of different
+    output channels (stdout, stderr) and the return code of the execution.
+
+    Delegates iterators, direct comparisons and unknown attribute retrieval
+    (including function calls) to the `stdout` attribute for better testing
+    ergonomics.
+    """
+    def __init__(self, retcode, stdout, stderr):
+        self.retcode = retcode
+        self.stdout = stdout
+        self.stderr = stderr
+
+    def __eq__(self, other):
+        if isinstance(other, str):
+            return self.stdout == other
+        elif isinstance(other, self.__class__):
+            return self.retcode == other.retcode and self.stdout == other.stdout and self.stderr == other.stderr
+        return False
+
+    def __contains__(self, other):
+        if isinstance(other, str):
+            return self.stdout.__contains__(other)
+        elif isinstance(other, self.__class__):
+            return self.retcode == other.retcode and self.stdout.__contains__(other.stdout) and self.stderr.__contains__(other.stderr)
+        return False
+
+    def __iter__(self):
+        return self.stdout.__iter__()
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}({self.retcode!r}, {self.stdout!r}, {self.stderr!r})'
+
+    def __str__(self):
+        return self.stdout
+
+    def __int__(self):
+        return int(self.__str__())
```

Comment:
I'm not yet sure about the intuition in this way though. I'm fine keeping it as is, but I see two different intuitions here:

- `int(aCommandResult)` returns `self.retcode`
- `int(aCommandResult)` returns `int(self.__str__())`, what is effectively `int(self.stdout)`

I would expect the first behavior I think, and users that want to parse the `stdout` to an int could do `int(aCommandResult.stdout)`.

Reply:
I'm personally inclined to expect the behavior of `int` to be stdout as an integer, just because "object treated as scalar value" behaves as a string containing stdout. While the proposed idea of using the return code does make sense, it seems like it might be non-intuitive to have a string become a completely different value when wrapped in a certain method.

I view `bool` as a different case, since that'd usually not be explicitly wrapped by a method; it'd usually be `if shellcommand()... `, I'd think. I know it's the same argument, but I think the feeling people have about program execution succeeding or failing is strong enough that having a different value makes sense.

But in both cases, I might just be projecting my personal thoughts onto everyone - which could be completely wrong. :D